tests/unit/delegate.ts

```diff
@@ -0,0 +1,93 @@
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import { emit } from 'dojo-core/on';
+import { Handle } from 'dojo-core/interfaces';
+import delegate from 'src/delegate';
+import * as dom from 'src/dom';
+
+let container = document.createElement('div');
+let handles: (Handle)[] = [];
+
+document.body.appendChild(container);
+
+function testDelegate(...args: any[]) {
+	let handle = delegate.apply(null, arguments);
+	handles.push(handle);
+	return handle;
+}
+
+registerSuite({
+	name: 'delegate',
+
+	afterEach() {
+		while (handles.length > 0) {
+			handles.pop().destroy();
+		}
+		container.innerHTML = '';
+	},
+
+	'CSS selector matching multiple elements'() {
+		let called = 0;
+		let buttonOne = document.createElement('button');
+		let buttonTwo = document.createElement('button');
+
+		delegate(container, 'button', 'click', function() {
+			called++;
+		});
+
+		container.appendChild(buttonOne);
+		container.appendChild(buttonTwo);
+
+		buttonOne.click();
+		buttonTwo.click();
+		assert.strictEqual(called, 2);
+	},
+
+	'CSS selector and text node target'() {
```

Comment:
Not really sure why this says "text node target" since AFAICT we're emitting off of a div element?